backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/AbstractElasticsearchIndexFieldTypeOptionsStep.java

```diff
@@ -40,6 +41,13 @@ public <V> S projectionConverter(Class<V> valueType, FromDocumentValueConverter<
 		return thisAsS();
 	}
 
+	@Override
+	public S queryStringConverter(ToDocumentValueConverter<String, ? extends F> fromQueryStringConverter) {
+		Contracts.assertNotNull( fromQueryStringConverter, "fromQueryStringConverter" );
+		builder.queryStringConverter( fromQueryStringConverter );
+		return thisAsS();
+	}
+
```

Comment:
This converter doesn't make much sense in the case of an Elasticsearch backend... but I wasn't sure what to do about it if we were going this route of using a converter...

Reply:
Good point... And making this Lucene-specific isn't an option either, since it would make integration with value bridges much harder...

However, I see you added `ValueConvert.PARSE` though, which means this converter is not only about query strings: it can be used for many predicates, including in particular the `match` predicate; and in that case, it can be used even with Elasticsearch./

So:

1. Shouldn't this method be named something like `parser(ToDocumentValueConverter<String, ? extends F>)` instead?
2. Shouldn't there be tests for `ValueConvert.PARSE` in the TCK for other predicates? See in particular `AbstractPredicateTypeCheckingAndConversionIT`.